packages/core/admin/ee/admin/content-manager/pages/EditView/InformationBox/InformationBoxEE.js

```diff
@@ -1,47 +1,93 @@
+  } = useCMEditViewDataManager();
+  const { put } = useFetchClient();
+  const activeWorkflowStage = initialData?.[ATTRIBUTE_NAME] ?? null;
   const { formatMessage } = useIntl();
+  const { formatAPIError } = useAPIErrorHandler();
 
-  const activeWorkflowStage = initialData?.[ATTRIBUTE_NAME] ?? null;
   const {
     workflows: { data: workflow },
-  } = useReviewWorkflows(1);
-  // stages are empty while the workflow is loaded
+  } = useReviewWorkflows(activeWorkflowStage?.id);
+
+  const { error, isLoading, mutateAsync } = useMutation(async ({ entityId, stageId, uid }) => {
+    const {
+      data: { data },
+    } = await put(`/admin/content-manager/collection-types/${uid}/${entityId}/stage`, {
+      data: { id: stageId },
+    });
+
+    return data;
+  });
+
+  // stages are empty while the workflow is loading
   const options = (workflow?.stages ?? []).map(({ id, name }) => ({ value: id, label: name }));
+  const formattedError = error ? formatAPIError(error) : null;
+
+  const handleStageChange = async ({ value: stageId }) => {
+    try {
+      await mutateAsync({
+        entityId: initialData.id,
+        stageId,
+        uid,
+      });
+    } catch (error) {
+      // react-query@v3: the error doesn't have to be handled here
+      // see: https://github.com/TanStack/query/issues/121
+    }
+  };
 
   return (
     <Information.Root>
       <Information.Title />
+
       {activeWorkflowStage && (
-        <Flex direction="column" gap={2} alignItems="stretch">
-          <FieldLabel htmlFor={ATTRIBUTE_NAME}>
-            {formatMessage({
-              id: 'content-manager.reviewWorkflows.stage.label',
-              defaultMessage: 'Review stage',
-            })}
-          </FieldLabel>
-
-          <ReactSelect
-            inputId={ATTRIBUTE_NAME}
-            isDisabled={isCreatingEntry}
-            options={options}
-            name={ATTRIBUTE_NAME}
-            defaultValue={{ value: activeWorkflowStage?.id, label: activeWorkflowStage?.name }}
-            isSearchable={false}
-            isClearable={false}
-          />
-        </Flex>
+        <Field error={formattedError} name={ATTRIBUTE_NAME}>
+          <Flex direction="column" gap={2} alignItems="stretch">
+            <FieldLabel>
+              {formatMessage({
+                id: 'content-manager.reviewWorkflows.stage.label',
+                defaultMessage: 'Review stage',
+              })}
+            </FieldLabel>
+
+            <ReactSelect
+              error={formattedError}
+              inputId={ATTRIBUTE_NAME}
+              isDisabled={isCreatingEntry}
+              options={options}
+              name={ATTRIBUTE_NAME}
+              defaultValue={{ value: activeWorkflowStage?.id, label: activeWorkflowStage?.name }}
+              isLoading={isLoading}
+              isSearchable={false}
+              isClearable={false}
+              onChange={handleStageChange}
+            />
+
+            {error && <FieldError />}
```

Comment:
you don't need to conditionally render this, it renders `null` if there's no error?